executor/common_kvm_arm64_syzos.h

```diff
@@ -11,10 +11,69 @@
 // Start/end of the guest section.
 extern char *__start_guest, *__stop_guest;
 
+#define dmb() asm volatile("dmb sy" \
+			   :        \
+			   :        \
+			   : "memory")
+
+#define SYZOS_CALL5(id, arg1, arg2, arg3, arg4, arg5)                                                                         \
+	do {                                                                                                                  \
+		dmb();                                                                                                        \
+		uint64 fn = ((uint64*)ARM64_ADDR_API_TRAMPOLINES)[id];                                                        \
+		dmb();                                                                                                        \
+		void (*api)(uint64, uint64, uint64, uint64, uint64) = (void (*)(uint64, uint64, uint64, uint64, uint64))(fn); \
+		(api)(arg1, arg2, arg3, arg4, arg5);                                                                          \
+	} while (0)
+
+typedef enum {
+	SYZOS_API_STOP = 0,
+	SYZOS_API_UCALL,
+	SYZOS_API_LAST, // Must be the last one
+} syzos_api_id;
+
+struct api_call {
+	uint64 call;
+	uint64 args[7];
+};
+
+GUEST_CODE void interpret_payload(uint64 addr, uint64 size)
+{
+	struct api_call* cmd = (struct api_call*)addr;
+	do {
+		if ((cmd->call == SYZOS_API_STOP) || (cmd->call >= SYZOS_API_LAST))
+			return;
```

Comment:
In fact we do that in an upcoming patch that will reserve some ucall numbers.
I think it's also fine as is, because with the instruction-based payload we aren't even sure it ever ends.